todo_app/app.py

```diff
@@ -1,15 +1,35 @@
-from flask import Flask
+from flask import Flask, render_template, request
 
 from todo_app.flask_config import Config
 
+from todo_app.data.session_items import add_item, delete_item, get_items, save_item
+
 app = Flask(__name__)
 app.config.from_object(Config)
 
 
 @app.route('/')
 def index():
-    return 'Hello World!'
+    items = sorted(get_items(), key=lambda item: item['status'], reverse=True)
+    return render_template('index.html', items=items)
+
+@app.route('/todo/add', methods = ['POST'])
+def create_new_todo():
+    title = request.form['title']
+    return add_item(title)
+
+@app.route('/todo/<itemId>/status', methods = ['PUT'])
+def change_state(itemId):
+    item = list(filter(lambda item: item['id'] == int(itemId), get_items())).pop()
```

Comment:
Nice use of doing this functionally - rather than converting to a list and then popping the first element, here we could take advantage of python's built in `next` function  which achieves the same e.g.
`item = next(filter(lambda item: item['id'] == int(itemId), get_items()))`

Reply:
Ah nice, have made that change.